### DISCUSSION TITLE
upgraded gcc on macos monterey, now says version 11.2.0_1 and can't compile

### DISCUSSION THREAD

**Discussion by adrio96** [Homebrew/discussions#2439] [Everyday usage] [answered]
I upgraded gcc on macos monterey, the version changed to 11.2.0_1 and now when I try compiling with gfortran I get the error:
ld: library not found for -lSystem 
Can someone tell me how to fix this? Is there a way to get the 11.2.0 version back? Does downgrading to macos bigsur and then reinstalling gcc solve this? Is there an easier way?

**Comment by SMillerDev** [comment]
What do `brew config` and `brew doctor` say?

**Reply by adrio96** [reply]
I don't know, I solved by uninstalling the gcc version installed with homebrew and installing this one:
https://github.com/fxcoudert/gfortran-for-macOS/releases

PS it's my first time here, how do I attach labels to the question?

**Comment by adrio96** [comment, marked as answer]
SOLVED by uninstalling the gcc version installed with homebrew and installing this one:
https://github.com/fxcoudert/gfortran-for-macOS/releases

**Reply by carlocab** [reply]
@fxcoudert is also a Homebrew maintainer, so their Gfortran build is probably the same as the GCC build in Homebrew/core. It's pretty odd that you're seeing the linker error with Homebrew GCC but not that version of Gfortran.

**Reply by fxcoudert** [reply]
I can confirm there is no fundamental difference between the GCC in Homebrew and that in my repo (except in the installing method). Without further details, it's impossible to know what the problem is in the original post.

**Reply by rouson** [reply]
I'm experiencing the same problem.  I installed macOS Monterey about 10 days ago.  I'm pretty confident that my `gcc` and `gfortran` worked afterward and I'm not sure what changed, but I'm glad to provide any information that would help in diagnosing this issue. I'd like to stick with Homebrew rather than installing another way because I'm writing an installation script for a package that has several other dependencies and the more things that I can install with one `brew` command, the simpler the script is.

**Reply by rouson** [reply]
My `gfortran` version is 11.2.0_3.

**Reply by rouson** [reply]
After hours of searching, I was finally in the process of submitting a Homebrew issue when I found that the issue template requires the output of `brew config` and `brew doctor`, that latter of which provided the steps that fixed this problem for me:
```
softwareupdate --all --install --force
sudo rm -rf /Library/Developer/CommandLineTools
sudo xcode-select --install
```
My XCode Command Line Tools installations were too old.
